library/src/main/scala/scala/scalajs/LinkingInfo.scala

```diff
@@ -108,4 +108,35 @@ object LinkingInfo {
   def assumingES6: Boolean =
     linkingInfo.assumingES6
 
+  /** The value of the global JavaScript `this`.
+   *
+   *  This returns the value that would be obtained by writing `this` at the
+   *  top-level of the JavaScript file generated by Scala.js. In most
+   *  JavaScript environments, this is equivalent to the *global object*, but
+   *  it is not necessarily the case. For example, on Node.js, `this` is the
+   *  object representing the `exports` of the current module.
+   *
+   *  Using this value should be rare, and mostly limited to writing code
+   *  detecting what the global object is. For example, a typical detection
+   *  code looks like:
+   *  {{{
+   *  val globalObject = {
+   *    import js.Dynamic.{global => g}
+   *    if (js.typeOf(g.global) != "undefined" && (g.global.Object eq g.Object)) {
+   *      // Node.js environment detected
+   *      g.global
+   *    } else {
+   *      // In all other well-known environment, we can use the global `this`
+   *      scala.scalajs.LinkingInfo.globalThis
+   *    }
+   *  }
+   *  }}}
+   *  Note that the above code is not comprehensive, as there can be JavaScript
+   *  environments where the global object cannot be fetched neither through
+   *  `global` nor `this`. If your code needs to run in such an environment, it
+   *  is up to you to use an appropriate detection procedure.
+   */
+  @inline
+  def globalThis: js.Dynamic = linkingInfo.globalThis
+
```

Comment:
> I don't fully understand this. AFAICS, there is currently no public EnvInfo, only a facade type for documentation (that does not even appear in the doc anymore).

Right, I meant the private facade type one. But in a sense, if that one disappears, it seems weird we would add now a public `EnvInfo`, which by all expectations will only ever contain `globalThis`.

> I thought this as well for quite a while, but is it really that bad?

I guess not, given that we also have `js.debugger()` there anyway 🙄 

> As I am thinking of this, I start wondering: Should global itself not be on js itself? I would argue that no, because it makes it clearer that code is dynamically typed.

I agree that `global` should not be in `js.global`. `global` is the entry point to the dynamically typed world. It can and should be avoided if the code is completely statically typed.

`globalThis` is slightly different in that respect. Even though it is typed as `js.Dynamic` (and now I'm wondering whether it should be `js.Object` instead, or even `scala.Any`), using it or not is orthogonal to living in a statically or dynamically world. If you need it, you need it, regardless of your typing strategy. That's why I would not put it in `js.Dynamic`.

> There is another option I just thought of: We could start an package js.completeness

I don't like the name but I see the point. Some things that could go there:

* `globalThis`
* `debugger()`
* `delete` (currently in `js.Dictionary`, but that's wrong)
* The primitive for `new.target` #2638.